scripts/publish.sh

```diff
@@ -0,0 +1,24 @@
+#!/usr/bin/env bash
+
+set -x
+set -e
+set -o pipefail
+
+if [[ -z $NPM_TOKEN ]]; then
+  echo "Notice: NPM_TOKEN environment variable not set. Running 'npm publish --dry-run'."
+  npm publish --dry-run
+  exit 0
+fi
+
+# check param, if it's set (monorepo) we check if it's published before proceeding
+if [[ -n "$1" ]]; then
+  # check if module is published
+  LATEST_VERSION=$(npm view . version || echo "")
+  CURRENT_VERSION=$(jq --raw-output .version package.json)
+  if [ "$LATEST_VERSION" == "$CURRENT_VERSION" ]; then
+    echo "Notice: This module is already published at this version. aborting publish."
+    exit 0
```

Comment:
This will not fail the build for a polyrepo. Is that what we want?

Reply:
we're only passing `true` to the publish script for the monorepo case so for polyrepo this whole block will get skipped and things will function as they did before.

Reply:
@mcmire I've actually gone ahead and added two new tests to make this a bit clearer